src/backend/feed/worker.js

```diff
@@ -1,19 +1,48 @@
-const feedparser = require('./parser');
 const feedQueue = require('./queue');
+const feedParser = require('./parser');
+const extractUrls = require('../utils/extract-urls');
+const storage = require('../utils/storage');
+// const logger = require('./../utils/logger');
 
 exports.workerCallback = async function(job) {
   const { url } = job.data;
-  const posts = await feedparser(url);
-  return posts.map(post => ({
-    author: post.author,
-    date: post.date,
-    title: post.title,
-    description: post.description,
-    postURL: post.link,
-  }));
+  try {
+    const posts = await feedParser(url);
+    const processedPosts = [];
+    if (posts.length > 0) {
+      posts.forEach(post => {
+        // We can extract any other information from the post that we need here.
+        const processedPost = {
+          author: post.author,
+          date: post.date,
+          title: post.title,
+          description: post.description,
+          postURL: post.link,
+        };
+        processedPosts.push(processedPost);
+        extractUrls.extract(post.description);
+      });
+      return processedPosts;
+    }
+    return processedPosts;
+    // return processedPosts;
+  } catch (err) {
+    return err;
+  }
 };
 
+// eslint-disable-next-line prettier/prettier
 exports.start = function() {
   // Start processing jobs from the feed queue...
-  feedQueue.process(this.workerCallback);
+  feedQueue.process(exports.workerCallback);
+  feedQueue.on('completed', (job, results) => {
+    if (results.length > 0) {
+      results.forEach(result => {
+        console.log(`Result: ${result.title}`);
+        // We have an issure here, addPost is an async function. Should we make the start
```

Comment:
Yes, we also need to update how you do this "loop" when you have promises for each:

```js
await Promise.all(results.map(async result => storate.addPost(result)))
```

And yeah, we should probably make `start` async.